include/fmt/base.h

```diff
@@ -1971,12 +1971,32 @@ template <typename T = char> class counting_buffer : public buffer<T> {
 template <typename T>
 struct is_back_insert_iterator<basic_appender<T>> : std::true_type {};
 
+template <typename T, typename It, typename = void>
+struct has_container_append : std::false_type {};
+template <typename T, typename It>
+struct has_container_append<T, It,
+                            void_t<decltype(std::declval<T>().append(
+                                std::declval<It>(), std::declval<It>()))>>
+    : std::true_type {};
+
 // An optimized version of std::copy with the output value type (T).
+template <typename T, typename It,
+          FMT_ENABLE_IF(has_container_append<T, It>::value)>
+FMT_CONSTEXPR20 void container_append(T& container, It begin, It end) {
+  container.append(begin, end);
+}
+
+template <typename T, typename It,
+          FMT_ENABLE_IF(!has_container_append<T, It>::value)>
+FMT_CONSTEXPR20 void container_append(T& container, It begin, It end) {
+  container.insert(container.end(), begin, end);
+}
```

Comment:
I don't think we need an extra "level of indirection". Let's merge these into `copy`.

Reply:
Merged